src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsSourceView.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml.Interop;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	/// <summary>
+	/// This implementation combines ItemsSourceView with InspectingDataSource to match behavior
+	/// </summary>
+	public class ItemsSourceView : INotifyCollectionChanged
+	{
+		private int m_cachedSize = -1;
+		private IList m_vector = null;
+		private IKeyIndexMapping m_uniqueIdMaping;
+
+		public ItemsSourceView(object source)
+		{
+			if (source == null)
+			{
+				throw new ArgumentNullException("Argument 'source' is null.");
+			}
+
+			var vector = source as IList<object>;
+			if (vector != null)
+			{
+				m_vector = vector;
+				ListenToCollectionChanges();
+			}
+			else
+			{
+				// The bindable interop interface are abi compatible with the corresponding
+				// WinRT interfaces.
+				var bindableVector = source as IList;
+				if (bindableVector != null)
+				{
+					m_vector.set(reinterpret_cast <const IVector<object>&> (bindableVector));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)